src/ray/core_worker/lib/java/jni_utils.h

```diff
@@ -571,13 +575,29 @@ inline jobject NativeRayObjectToJavaNativeRayObject(
   if (!rayObject) {
     return nullptr;
   }
-  auto java_data = NativeBufferToJavaByteArray(env, rayObject->GetData());
+  auto metadata_ptr = rayObject->GetMetadata();
+  auto metadata_str = std::string(reinterpret_cast<const char *>(metadata_ptr->Data()),
+                                  metadata_ptr->Size());
   auto java_metadata = NativeBufferToJavaByteArray(env, rayObject->GetMetadata());
-  auto java_obj = env->NewObject(java_native_ray_object_class,
-                                 java_native_ray_object_init, java_data, java_metadata);
-  RAY_CHECK_JAVA_EXCEPTION(env);
+  jobject java_obj;
+  if (metadata_str.rfind(OBJECT_METADATA_TYPE_ARROW) == 0) {
+    auto data_ptr = rayObject->GetData();
+    auto java_data = env->NewDirectByteBuffer(
```

Comment:
Why we not just put data to NativeRayObject.data instead introducing the DirectBuffer? I don't think it does matter that the data is in JVM heap or native memory.

Reply:
Using direct byte buffer can avoid first copying the data in RayObject to heap. This way java read performance is better.

Reply:
I see. But I still don't know what's the perf gap here indeed. If it's necessary to do this, are you willing to  submit another PR to make NativeRayObject.data as a DirectBuffer as well?

Reply:
OK,